api_app/analyzers_manager/observable_analyzers/phunter/phunter_base.py

```diff
@@ -0,0 +1,72 @@
+import phonenumbers
+import requests
+from bs4 import BeautifulSoup
+from phonenumbers import carrier
+
+
+class PhunterBase:
+    def __init__(self):
+        pass
+
+    @staticmethod
+    def phunt(phone_number: str) -> dict:
+
+        # General Information
+
+        parsed = phonenumbers.parse(phone_number)
+
+        possible = phonenumbers.is_possible_number(parsed)
+
+        valid = phonenumbers.is_valid_number(parsed)
+
+        operator = carrier.name_for_number(parsed, "en")
+        if operator == "":
+            operator = "Not found"
+
+        line = phonenumbers.number_type(parsed)
+        if line == phonenumbers.PhoneNumberType.FIXED_LINE:
+            ligne = "Fixed"
+        elif line == phonenumbers.PhoneNumberType.MOBILE:
+            ligne = "Mobile"
+        else:
+            ligne = "Not found"
+
+        # Free Lookup
+
+        user_agent = "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:124.0) Gecko/20100101 Firefox/124.0"
+
+        free_lookup_url = f"https://free-lookup.net/{phone_number.replace('+', '')}"
+
+        r = requests.get(free_lookup_url, headers={"user-agent": user_agent})
+
+        html_body = BeautifulSoup(r.text, "html.parser")
+        list_info = html_body.findChild("ul", class_="report-summary__list").findAll(
+            "div"
+        )
+
+        info_dict = {
+            k.text.strip(): info.text.strip() if info.text.strip() else "Not found"
+            for _, (k, info) in enumerate(zip(list_info[::2], list_info[1::2]))
+        }
```

Comment:
Please wrap this in a separated function and inside a try/except clause. In this way, if this part, for any reason, fails, the other analyzer parts still get the results to the users